javaServices/coreJavaServices/src/joliex/util/TimeService.java

```diff
@@ -415,4 +422,38 @@ public Long getTimestampFromString( Value request )
 			throw new FaultException( "InvalidTimestamp", pe );
 		}
 	}
+
+	@RequestResponse
+	public String scheduleTimeout( Value request ) {
+		final String uuid = UUID.randomUUID().toString();
+		TimeUnit unit;
+
+		if (request.getFirstChild("timeunit") != null) {
+			try {
+				unit = TimeUnit.valueOf(request.getFirstChild("timeunit").strValue().toUpperCase());
+			} catch (Exception e) {
+				System.err.println(e);
+				System.err.println("Defaulting to timeunit seconds");
+			}
+			finally {
+				unit = TimeUnit.SECONDS;
+			}
+		} else {
+			unit = TimeUnit.SECONDS;
+		}
+
+		tasks.add( uuid );
+		executor.schedule( () -> {
+			if ( tasks.contains( uuid ) ) {
+				sendMessage(CommMessage.createRequest(request.getFirstChild("operation").strValue(), "/", request.getFirstChild("message")));
+				tasks.remove(uuid);
+			}
+		}, request.intValue(), unit );
+		return uuid;
+	}
+
+	public void cancelTimeout( Value request ) {
+		tasks.remove( request.strValue() );
```

Comment:
But the task is still running in the executor, which could prevent the Jolie program from terminating in case we're done with the service. `executor.schedule` actually returns a `ScheduleFuture` that supports a `cancel` method that you can use to cancel the task.